Update Set-CsTeamsAppPermissionPolicy.md (#10675)
MicrosoftDocs/office-docs-powershell: skype/skype-ps/skype/Set-CsTeamsAppPermissionPolicy.md

## EXAMPLES

### Example 1

> No apps ae blocked,allow all Microsoft apps,Third-party apps,Custom apps.

Reviewer: ```suggestion
This example allows all Microsoft apps, Third-party apps, and Custom apps. No apps are blocked.

```

> No apps ae blocked,allow all Microsoft apps,Third-party apps,Custom apps.

Reviewer 2: Please move the description below the code

Author: got it,thanks！

```powershell
$identity = "TestTeamsAppPermissionPolicy" + (Get-Date -Format FileDateTimeUniversal)
New-CsTeamsAppPermissionPolicy -Identity Set-$identity
Set-CsTeamsAppPermissionPolicy -Identity Set-$identity -DefaultCatalogAppsType BlockedAppList  -DefaultCatalogApps @()-GlobalCatalogAppsType -GlobalCatalogApps @() BlockedAppList -PrivateCatalogAppsType BlockedAppList -PrivateCatalogApps @()
```

### Example 2

> No apps are allowed,block all Microsoft apps,Third-party apps,Custom apps.

Reviewer: ```suggestion
This example blocks all Microsoft apps, Third-party apps, and Custom apps. No apps are allowed.

```

```powershell
$identity = "TestTeamsAppPermissionPolicy" + (Get-Date -Format FileDateTimeUniversal)
New-CsTeamsAppPermissionPolicy -Identity Set-$identity
Set-CsTeamsAppPermissionPolicy -Identity Set-$identity -DefaultCatalogAppsType AllowedAppList -DefaultCatalogApps @() -GlobalCatalogAppsType AllowedAppList -GlobalCatalogApps @() -PrivateCatalogAppsType AllowedAppList -PrivateCatalogApps @()
```

### Example 3

> Allow Microsoft Lists、OneNote apps that can be installed by your users and block other Microsoft apps.

Reviewer: ```suggestion
This example allows Microsoft Lists and OneNote apps and blocks other Microsoft apps. Microsoft Lists and OneNote can be installed by your users.

```

```

### Example 4

Allow Third-party TaskList、OnePlan apps that can be installed by your users and block other Third-party apps.
```powershell
$identity = "TestTeamsAppPermissionPolicy" + (Get-Date -Format FileDateTimeUniversal)
# create a new Teams app permission policy and block all apps
New-CsTeamsAppPermissionPolicy -Identity Set-$identity -DefaultCatalogAppsType AllowedAppList -GlobalCatalogAppsType AllowedAppList -PrivateCatalogAppsType AllowedAppList -DefaultCatalogApps @() -GlobalCatalogApps @() -PrivateCatalogApps @()
$TaskListApp = New-Object -TypeName Microsoft.Teams.Policy.Administration.Cmdlets.Core.GlobalCatalogApp -Property @{Id="57c81e84-9b7b-4783-be4e-0b7ffc0719af"}
$OnePlanApp = New-Object -TypeName Microsoft.Teams.Policy.Administration.Cmdlets.Core.GlobalCatalogApp -Property @{Id="ca0540bf-6b61-3027-6313-a7cb4470bf1b"}
$GlobalCatalogAppList = @($TaskListApp,$OnePlanApp)
# set allow TaskList and OnePlan apps and block other Third-party apps
Set-CsTeamsAppPermissionPolicy -Identity Set-$identity -GlobalCatalogAppsType AllowedAppList  -GlobalCatalogApps $GlobalCatalogAppList
```

### Example 5

> Allow Custom GetStartApp、TestBotApp apps that can be installed by your users and block other Custom apps.

Reviewer: ```suggestion
This example allows Custom GetStartApp and TestBotApp apps and blocks other Custom apps. GetStartApp and TestBotApp can be installed by your users.

```

## EXAMPLES

### Example 1

```powershell
$identity = "TestTeamsAppPermissionPolicy" + (Get-Date -Format FileDateTimeUniversal)
New-CsTeamsAppPermissionPolicy -Identity Set-$identity
Set-CsTeamsAppPermissionPolicy -Identity Set-$identity -DefaultCatalogAppsType BlockedAppList  -DefaultCatalogApps @()-GlobalCatalogAppsType -GlobalCatalogApps @() BlockedAppList -PrivateCatalogAppsType BlockedAppList -PrivateCatalogApps @()
```

> This example allows all Microsoft apps, Third-party apps, and Custom apps. No apps are blocked.

Reviewer 3: ```suggestion
This example allows all Microsoft apps, third-party apps, and custom apps. No apps are blocked.

```

### Example 2

```powershell
$identity = "TestTeamsAppPermissionPolicy" + (Get-Date -Format FileDateTimeUniversal)
New-CsTeamsAppPermissionPolicy -Identity Set-$identity
Set-CsTeamsAppPermissionPolicy -Identity Set-$identity -DefaultCatalogAppsType AllowedAppList -DefaultCatalogApps @() -GlobalCatalogAppsType AllowedAppList -GlobalCatalogApps @() -PrivateCatalogAppsType AllowedAppList -PrivateCatalogApps @()
```

> This example blocks all Microsoft apps, Third-party apps, and Custom apps. No apps are allowed.

Reviewer 3: ```suggestion
This example blocks all Microsoft apps, third-party apps, and custom apps. No apps are allowed.

```

```
This example allows Microsoft Lists and OneNote apps and blocks other Microsoft apps. Microsoft Lists and OneNote can be installed by your users.

### Example 4

```powershell
$identity = "TestTeamsAppPermissionPolicy" + (Get-Date -Format FileDateTimeUniversal)
# create a new Teams app permission policy and block all apps
New-CsTeamsAppPermissionPolicy -Identity Set-$identity -DefaultCatalogAppsType AllowedAppList -GlobalCatalogAppsType AllowedAppList -PrivateCatalogAppsType AllowedAppList -DefaultCatalogApps @() -GlobalCatalogApps @() -PrivateCatalogApps @()
$TaskListApp = New-Object -TypeName Microsoft.Teams.Policy.Administration.Cmdlets.Core.GlobalCatalogApp -Property @{Id="57c81e84-9b7b-4783-be4e-0b7ffc0719af"}
$OnePlanApp = New-Object -TypeName Microsoft.Teams.Policy.Administration.Cmdlets.Core.GlobalCatalogApp -Property @{Id="ca0540bf-6b61-3027-6313-a7cb4470bf1b"}
$GlobalCatalogAppList = @($TaskListApp,$OnePlanApp)
# set allow TaskList and OnePlan apps and block other Third-party apps
Set-CsTeamsAppPermissionPolicy -Identity Set-$identity -GlobalCatalogAppsType AllowedAppList  -GlobalCatalogApps $GlobalCatalogAppList
```

> This example allows Third-party TaskList and OnePlan apps and blocks other Third-party apps.  TaskList and OnePlan can be installed by your users.

Reviewer 3: ```suggestion
This example allows third-party TaskList and OnePlan apps and blocks other third-party apps.  TaskList and OnePlan can be installed by your users.

```

```

### -DefaultCatalogAppsType

> Choose to allow or block the installation of Microsoft Apps.

Reviewer 3: ```suggestion
Choose to allow or block the installation of Microsoft apps.

```

```

### -DefaultCatalogAppsType
Choose to allow or block the installation of Microsoft apps.

```yaml
Type: String

> Parameter Sets: AllowedAppList,BlockedAppList

Reviewer: ```suggestion
Parameter Sets: (All)
```

```

### -Description

> Description of App setup permission policy.

Reviewer 3: ```suggestion
Description of app setup permission policy.

```

```

### -GlobalCatalogApps

> Choose which Teams apps published by a third-party that can be installed by your users.

Reviewer 3: ```suggestion
Choose which Teams apps published by a third party can be installed by your users.

```

```

### -GlobalCatalogAppsType

> Choose to allow or block the installation of third-party.

Reviewer 3: ```suggestion
Choose to allow or block the installation of third-party apps.

```

```

### -GlobalCatalogAppsType
Choose to allow or block the installation of third-party apps.

```yaml
Type: String

> Parameter Sets: AllowedAppList,BlockedAppList

Reviewer 2: The parameters sets in the article are Identity and Instance, are these that you are mentioning here, valid parameter sets?

Author: yes, you can choose one of them as GlobalCatalogAppsType

Reviewer 2: One thing is a parameter set name and the other a parameter value

> Parameter Sets: AllowedAppList,BlockedAppList

Reviewer: ```suggestion
Parameter Sets: (All)
```

I agree with @get-itips. Parameter sets should be (All). AllowedAppList and BlockedAppList should be possible values in the description.

```

### -Identity

> Name of App setup permission policy.If empty, all Identities will be used by default

Reviewer: ```suggestion
Name of App setup permission policy. If empty, all Identities will be used by default.

```

```

### -PrivateCatalogAppsType
Choose which custom apps can be installed by your users.

```yaml
Type: String

> Parameter Sets: AllowedAppList,BlockedAppList

Reviewer: ```suggestion
Parameter Sets: (All)
```
